### PR TITLE
feat: add specific log file to gather current patch-hub execution logs 

### LINKED ISSUE #58: Write current patch-hub execution logs to `latest.log` file
### **Description**:

All logs from the current patch-hub execution should be written to a specific file. The file should retain its current logs until another patch-hub execution begins. Additionally, the current behavior of saving each execution log in a separate file to preserve log history should remain unchanged.

### **Motivation**:

Currently, for each patch-hub run, we save a separate log file. This is great for preserving the log history of old runs but not ideal for debugging patch-hub in real-time. A simple solution to make debugging easier would be to have an additional log file with a fixed path, like latest.log. This file would always contain the logs of the current patch-hub run, allowing developers to open another terminal and use a command like watch -n 0.5 -d cat latest.log to see the logs as they are generated.

### PR DESCRIPTION
**Pull request by lorenzoberts**
This PR makes it easier to debug patch-hub since the developer can always get the current logs by looking at the same file, allowing them to follow the logs as they are generated.

In the future, we could create a specific cargo command to watch the` latest.log` file, but for now, we need to run a command like: `watch -n 0.5 -d cat ~/.local/share/patch_hub/logs/latest.log`.

Thanks to @OJarrisonn for the file name suggestion.

Closes #58

### CODE REVIEW

**Comment by OJarrisonn** [comment]
If i'm not mistaken, `tail -f` also can be used to read the log file while it's being written

**Comment by davidbtadokoro** [comment]
> If i'm not mistaken, `tail -f` also can be used to read the log file while it's being written

True and maybe more simple, as `watch` overwrites the whole screen. So in the case of many events, the visualization would be cleaner with `tail -f`.

File: src/app/logging.rs [hunk 1]
```diff
@@ -58,6 +62,8 @@ pub struct LogMessage {
 pub struct Logger {
     log_file: Option<File>,
     log_filepath: Option<String>,
+    latest_log_file: Option<File>,
+    latest_log_filepath: Option<String>,
```
Comment (davidbtadokoro) [inline src/app/logging.rs:66]: At first, I thought this field was unnecessary, as it was only used when creating the latest log file in `init_log_file()`. This prompted me to think that the field `log_filepath` was also useless, but I realized that @OJarrisonn did this as this field is used in `flush()`.

TLDR: shouldn't we also flush the latest log like we do with a normal log? Files aren't buffered in Rust and are written straightly (otherwise, we couldn't watch the file), but to maintain the pattern and be as safe as possible, maybe we need to enforce flushing here to.

File: src/main.rs [hunk 2]
```diff
@@ -23,6 +23,7 @@ fn main() -> color_eyre::Result<()> {
     run_app(&mut terminal, &mut app)?;
     utils::restore()?;
 
+    Logger::info("patch-hub finished");
```
Comment (davidbtadokoro) [inline src/main.rs:26]: Nitpick: this shouldn't pertain to this commit. However, I don't mind merging it this way :grimacing:

**Review by davidbtadokoro — COMMENTED** [review]
Nice PR @lorenzoberts! There's really one simple matter about the first comment I made.

In any case, I've tested the change in a handful of scenarios and it works as intended :+1:

File: src/app/logging.rs (hunk 1 above)
Comment (davidbtadokoro) [reply src/app/logging.rs:66]: Forget, I was mistaken about what I wrote.

**Comment by davidbtadokoro** [comment]
Change merged into the unstable branch :+1:
